be/src/olap/rowset/rowset.h

```diff
@@ -113,13 +113,26 @@ class Rowset : public std::enable_shared_from_this<Rowset> {
 
     virtual int64_t txn_id() const = 0;
     
+    // flag for push delete rowset
     virtual bool delete_flag() = 0;
 
     virtual bool check_path(const std::string& path) = 0;
 
+    bool need_delete_file() {
+        return _need_delete_file;
```

Comment:
if _need_delete_file = false and rowset shared_ptr use_count() == 1 will happen meanwhile?